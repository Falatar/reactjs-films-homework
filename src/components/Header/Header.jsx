import React, { Component } from 'react';
import PropTypes from 'prop-types';
import style from './Header.scss';
import Search from '../Search';
import FilmTitle from '../FilmTitle';
import Info from '../Info';

class Header extends Component {
  componentDidMount() {
    const { LoadMostPopularFilm } = this.props;
    LoadMostPopularFilm();
  }

  render() {
    const { mostPopularFilm } = this.props;
    if (mostPopularFilm.title) {
      return (
        <div
          className={style.header}
          id="space"
          style={{ backgroundImage: `url(https://image.tmdb.org/t/p/original${mostPopularFilm.backdrop_path})` }}
        >
          <Search />
          <div className={style.film__data}>
            <FilmTitle
              name={mostPopularFilm.title}
              genrePtime={mostPopularFilm.genre_str}
              rating={mostPopularFilm.vote_average}
            />
            <Info
              overview={mostPopularFilm.overview}
              id={mostPopularFilm.id}
            />
          </div>
        </div>
      );
    }
    return (
      <div className={style.header}>
        <span className={style.loading}>loading presentation...</span>
      </div>
    );
  }
}

Header.defaultProps = {
  LoadMostPopularFilm: () => {},
  mostPopularFilm: {},
};

Header.propTypes = {
  LoadMostPopularFilm: PropTypes.func,
  mostPopularFilm: PropTypes.objectOf(PropTypes.any),
};

export default Header;
