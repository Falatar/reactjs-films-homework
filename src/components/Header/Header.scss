.header {
    width: 100%;
    height: calc(100vw/2.3);
    border: 0;
    padding: 0;
    margin: 0;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-size: cover;
    background-position-y: center;
    box-shadow: inset 0 110px 150px  -50px black , inset 0 -110px 150px  -50px black;
    background-color: lightgrey;

    ::selection {
        background-color: dimgray;
    }

    .loading {
        color: white;
    }
}


.search__line, .film__data {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}

.film__data {
    height: 50%;
    align-items: flex-end;
}