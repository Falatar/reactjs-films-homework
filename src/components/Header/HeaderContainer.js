import { connect } from 'react-redux';
import loadTopFilm, { loadGenres } from '../../modules/filmListModule/filmListModuleActions';
import { getTopFilm, getGenres } from '../../modules/filmListModule/filmListModuleSelector';
import Header from './Header';

const mapStateToProps = (store) => ({
  mostPopularFilm: getTopFilm(store),
  genList: getGenres(store),
});

const mapDispatchToProps = (dispatch) => ({
  LoadMostPopularFilm: () => dispatch(loadTopFilm()),
  LoadGenreList: () => dispatch(loadGenres()),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Header);
