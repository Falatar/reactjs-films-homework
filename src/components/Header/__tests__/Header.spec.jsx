import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

jest.mock('../../Search', () => 'Search');
jest.mock('../../FilmTitle', () => 'FilmTitle');
jest.mock('../../Info', () => 'Info');

describe('Header', () => {
  const testInput = {
    backdrop_path: '',
    title: 'test',
    genre_str: '',
    vote_average: 0,
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<Header mostPopularFilm={testInput} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without input', () => {
    const tree = renderer
      .create(<Header mostPopularFilm={undefined} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
