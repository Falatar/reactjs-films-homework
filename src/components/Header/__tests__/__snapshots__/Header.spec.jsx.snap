// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="header"
  id="space"
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/original)",
    }
  }
>
  <Search />
  <div
    className="film__data"
  >
    <FilmTitle
      genrePtime=""
      name="test"
      rating={0}
    />
    <Info />
  </div>
</div>
`;

exports[`Header renders correctly without input 1`] = `
<div
  className="header"
>
  <span
    className="loading"
  >
    loading presentation...
  </span>
</div>
`;
